packages/tailwindcss/src/compile.ts

```diff
@@ -95,18 +95,19 @@ export function compileCandidates(
     // Find the first property that is different between the two rules
     let offset = 0
     while (
-      aSorting.properties.length < offset &&
-      zSorting.properties.length < offset &&
-      aSorting.properties[offset] === zSorting.properties[offset]
+      offset < aSorting.properties.order.length &&
+      offset < zSorting.properties.order.length &&
```

Comment:
This check was always wrong. `offset` started as `0` and no array has a length of `< 0` 😅